core/core/src/observability/mod.rs

```diff
@@ -0,0 +1,84 @@
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::{
+    filter::FilterFn, fmt::format::json, layer::SubscriberExt, EnvFilter,
+    Layer, util::SubscriberInitExt,
+};
+
+use self::buffer::{SharedEventBuffer, VecEventBuffer, RingEventBuffer};
+
+pub mod buffer;
+
+static mut METRICS_BUFFER: Option<SharedEventBuffer<VecEventBuffer>> = None;
+static mut DEVELOPER_DUMP_BUFFER: Option<SharedEventBuffer<RingEventBuffer>> = None;
+
+/// SAFETY: must only be called once during initialization of the program
+pub(crate) unsafe fn init(ring_event_buffer_size: usize) {
+    // SAFETY: this is only called once and there is no asynchronous mutation
+    unsafe {
+        METRICS_BUFFER.replace(SharedEventBuffer::new(VecEventBuffer::new()));
+        DEVELOPER_DUMP_BUFFER.replace(SharedEventBuffer::new(RingEventBuffer::new(ring_event_buffer_size)));
+
+        init_tracing(
+            METRICS_BUFFER.as_ref().cloned().unwrap(),
+            DEVELOPER_DUMP_BUFFER.as_ref().cloned().unwrap()
+        );
+    }
+}
+
+fn init_tracing(
+    metrics_buffer: SharedEventBuffer<VecEventBuffer>,
+    developer_dump_buffer: SharedEventBuffer<RingEventBuffer>
+) {
+    // we set up these layers:
+    // * user layer (@user) - output intended/relevant for users of the sdk
+    // * metrics layer (@metrics) - metrics sent to the dashboard
+    // * developer layer (everything) - output not relevant for normal users, but relevant when debugging and during development
+    // * dump layer (not @metrics) - output dumped after a panic, excluding metrics which are dumped separately
+
+    let user_layer = tracing_subscriber::fmt::layer()
+        .with_target(false)
+        .with_writer(std::io::stdout)
+        .with_filter(FilterFn::new(|metadata| {
+            metadata.target().starts_with("@user")
+        }))
+        .with_filter(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::WARN.into())
+                .with_env_var("SF_LOG")
```

Comment:
question: should it be ranamed to not mention reference sf as here https://github.com/superfaceai/one-sdk/pull/44?

Only trouble is, that I don't have good idea what the name shouold be. ONECLIENT_LOG feels unnecesarily long.

Reply:
Yes, I think it should. I am thinking `OSDK_LOG` and `OSDK_DEV_LOG`? And then there are also some `SF_CONFIG_*` ones and `SF_REGISTRY_URL`.

Reply:
Having it cross all tooling is god idea.

At the end it doesn't matter that much. It is important, it can be discovered by users.

Reply:
Renamed all variables and listed them in the readme. It should probably be in a separate PR but it's a small change imo.